tests/main/cgroup-pids/task.yaml

```diff
@@ -0,0 +1,57 @@
+summary: Each snap app and hook is tracked in a pids cgroup
+
+details: |
+    This test creates a snap process that suspends itself and ensures that it
+    placed into the appropriate hierarchy under the pids cgroup.
+
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+restore: |
+    rmdir /sys/fs/cgroup/pids/snap.test-snapd-sh.test-snapd-sh || true
+
+    # TODO: There is currently no way to unset configuration options.
+    # Once this is fixed please uncomment this line:
+    # snap unset core experimental.refresh-app-awareness
+
+execute: |
+    trap 'killall test-snapd-sh || true' EXIT
+    # Start a "sleep" process in the background
+    #shellcheck disable=SC2016
+    test-snapd-sh -c 'touch $SNAP_DATA/1.stamp && exec sleep 1h' &
```

Comment:
I wonder about this 1h arbitrary sleep. Should this be a while loop with a 'sleep 1h' in it? Should it be sleep 24h? Looking below, it should be ok...

Reply:
This is deliberately longer than the 50 minute timeout when tests fail. It's also an overkill since the test completes in seconds. We kill this command moments after starting it.